flows/vision-zero/cris_import/cris_import.py

```diff
@@ -546,59 +611,186 @@ def align_records(typed_token, dry_run):
             else:
                 # An insert is always just an vanilla insert, as there is not a pair of records to compare.
                 # Produce the SQL which creates a new VZDB record from a query of the imported data
-                insert_statement = util.form_insert_statement(output_map, table, input_column_names, import_key_sql, DB_IMPORT_SCHEMA)
+                insert_statement = util.form_insert_statement(output_map, table, input_column_names, import_key_sql, map_state["import_schema"])
                 logger.info(f"Executing insert in {output_map[table]} for where " + record_key_sql)
 
                 # Execute the insert statement
                 util.try_statement(pg, output_map, table, record_key_sql, insert_statement, dry_run)
 
     # fmt: on
-    return True
+    return map_state
+
+
+@task(
+    name="Group CSVs into logical groups",
+)
+def group_csvs_into_logical_groups(extracted_archives, dry_run):
+    files = os.listdir(str(extracted_archives))
+    logical_groups = []
+    for file in files:
+        if file.endswith(".xml"):
+            continue
+        match = re.search("^extract_(\d+_\d+)_", file)
+        group_id = match.group(1)
+        if group_id not in logical_groups:
+            logical_groups.append(group_id)
+    print("logical groups: " + str(logical_groups))
+    map_safe_state = []
+    for group in logical_groups:
+        map_safe_state.append({
+            "logical_group_id": group,
+            "working_directory": str(extracted_archives),
+            "csv_prefix": "extract_" + group + "_",
+            "dry_run": dry_run,
+        })
+    print(map_safe_state)
+    return map_safe_state
+
+
+@task(
+    name="Generate a short alphanumeric string based on logical group id",
+)
+def create_import_schema_name(mapped_state):
+    print(mapped_state)
+    schema = 'import_' + hashlib.md5(mapped_state["logical_group_id"].encode()).hexdigest()[:12]
+    mapped_state["import_schema"] = schema
+    print("Schema name: ", mapped_state["import_schema"])
+    return mapped_state
 
 
+@task(
+    name="Create target import schema",
+)
+def create_target_import_schema(map_state):
```

Comment:
Because we can't have multiple imports happening out of the same `import` schema, we're going to create a schema for every logical import group.